fipy/viewers/matplotlibViewer/matplotlib2DViewer.py

```diff
@@ -4,18 +4,37 @@
 
 from fipy.tools import numerix
 
-from fipy.viewers.matplotlibViewer.matplotlibViewer import AbstractMatplotlibViewer, _ColorBar
+from fipy.viewers.matplotlibViewer.matplotlibViewer import AbstractMatplotlibViewer
 
 __all__ = ["Matplotlib2DViewer"]
 from future.utils import text_to_native_str
 __all__ = [text_to_native_str(n) for n in __all__]
 
 class AbstractMatplotlib2DViewer(AbstractMatplotlibViewer):
-    def figaspect(self, figaspect):
+    def figaspect(self, figaspect, colorbar):
         if figaspect == 'auto':
             figaspect = self.vars[0].mesh.aspect2D
+            # We can't make the colorbar, yet (as we don't even have a figure)
+            # so hardcode these values, based on
+            # https://matplotlib.org/api/_as_gen/matplotlib.figure.Figure.html#matplotlib.figure.Figure.colorbar
+            #   fraction	0.15; fraction of original axes to use for colorbar
+            #   pad	        0.05 if vertical, 0.15 if horizontal; fraction of original axes between colorbar and new image axes
+            if colorbar == "vertical":
```

Comment:
Looks good, except using the variable name `colorbar` to represent an orientation, rather than something more intuitively colorbar-related, seems awkward.

Reply:
What should it represent? What should it be called?

I don't think an overdue bug fix is the place for a change to a 10 year old API.